backend/controller/leader/leader_test.go

```diff
@@ -0,0 +1,117 @@
+package leader
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/TBD54566975/ftl/backend/controller/leases"
+	"github.com/TBD54566975/ftl/internal/log"
+	"github.com/alecthomas/assert/v2"
+	"github.com/alecthomas/types/optional"
+)
+
+func TestExistingLeaseForURL(t *testing.T) {
+	// Test that if the a lease exists with the current URL, neither a leader or follower is created
```

Comment:
Buy "current" URL do you mean our own URL or something else?

Reply:
The lease has metadata set to the url of the leader. This is testing whether the coordinator's url matches the leader's url, but the coordinator does not have the lease, which can happen if a lease loses the connection, or a controller starts up at a url that was used by a controller that died.
I'll reword it, "current" isn't right.